tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -120,6 +121,111 @@ public async Task VerifyRedisResource()
+
+        var redis2 = builder2.AddRedis("redisForInsightPersistence")
+                .WithRedisInsight(c =>
+                {
+                    redisInsightBuilder = c;
+                    c.WithLifetime(ContainerLifetime.Persistent);
+                });
+
+        // Wire up an additional event subcription to ResourceReadyEvent on the RedisInsightResource
+        // instance. This works because the ResourceReadyEvent fires non-blocking sequential so the
+        // wire-up that WithRedisInsight does is guaranteed to execute before this one does. So we then
+        // use this to block pulling the list of databases until we know they've been updated. This
+        // will repeated below for the second app.
+        Assert.NotNull(redisInsightBuilder);
+        redisInsightsReady = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+        builder2.Eventing.Subscribe<ResourceReadyEvent>(redisInsightBuilder.Resource, (evt, ct) =>
+        {
+            redisInsightsReady.TrySetResult();
+            return Task.CompletedTask;
+        });
+
+        using var app2 = builder2.Build();
+        await app2.StartAsync(cts.Token);
+
+        await redisInsightsReady.Task.WaitAsync(cts.Token);
+
+        using var client2 = app2.CreateHttpClient($"{redisInsightBuilder.Resource.Name}", "http");
+        var secondRunDatabases = await client2.GetFromJsonAsync<RedisInsightDatabaseModel[]>("/api/databases", cts.Token);
+
+        Assert.NotNull(secondRunDatabases);
+        Assert.Single(secondRunDatabases);
+        Assert.Equal($"{redis2.Resource.Name}", secondRunDatabases[0].Name);
+        Assert.NotEqual(secondRunDatabases.Single().Id, firstRunDatabases.Single().Id);
+
+        // HACK: This is a workaround for the fact that ApplicationExecutor is not a public type. What I have
+        //       done here is I get the latest event from RNS for the insights instance which gives me the resource
+        //       name as known from a DCP perspective. I then use the ApplicationExecutorProxy (introduced with this
+        //       change to call the ApplicationExecutor stop method. The proxy is a public type with an internal
+        //       constructor inside the Aspire.Hosting.Tests package. This is a short term solution for 9.0 to
+        //       make sure that we have good test coverage for WithRedisInsight behavior, but we need a better
+        //       long term solution in 9.x for folks that will want to do things like execute commands against
+        //       resources to stop specific containers.
+        var rns = app2.Services.GetRequiredService<ResourceNotificationService>();
+        var latestEvent = await rns.WaitForResourceHealthyAsync(redisInsightBuilder.Resource.Name, cts.Token);
+        var executorProxy = app2.Services.GetRequiredService<ApplicationExecutorProxy>();
+        await executorProxy.StopResourceAsync(latestEvent.ResourceId, cts.Token);
```

Comment:
Can you get the `Stop` command for the resource and execute it? Simulating someone clicking the "Stop" button in the Dashboard.

Reply:
Similar problem. The command executor is internal. I've created an issue to look into this area in 9.x:

https://github.com/dotnet/aspire/issues/6455

Reply:
It feels like the command executor not being public is a bigger issue. How do I write functional tests for my commands?

cc @JamesNK